folding-schemes/src/folding/hypernova/mod.rs

```diff
@@ -25,3 +55,534 @@ impl<F: PrimeField> Witness<F> {
+                get_cm_coordinates(&self.U_i.C),
+                get_cm_coordinates(&self.u_i.C),
+                get_cm_coordinates(&U_i1.C),
+            ]
+            .concat();
+
+            let cf_circuit = CycleFoldCircuit::<C1, GC1> {
+                _gc: PhantomData,
+                r_bits: Some(rho_bits.clone()),
+                p1: Some(self.U_i.clone().C),
+                p2: Some(self.u_i.clone().C),
+                x: Some(cf_u_i_x.clone()),
+            };
+
+            let (_cf_w_i, cf_u_i, cf_W_i1, cf_U_i1, cf_cmT, _) =
+                fold_cyclefold_circuit::<C1, GC1, C2, GC2, FC, CS1, CS2>(
+                    &self.poseidon_config,
+                    self.cf_r1cs.clone(),
+                    self.cf_cs_params.clone(),
+                    self.cf_W_i.clone(), // CycleFold running instance witness
+                    self.cf_U_i.clone(), // CycleFold running instance
+                    cf_u_i_x,
+                    cf_circuit,
+                )?;
+
+            cf_u_i1_x = cf_U_i1.hash_cyclefold(&self.poseidon_config)?;
+
+            augmented_f_circuit = AugmentedFCircuit::<C1, C2, GC2, FC> {
+                _c2: PhantomData,
+                _gc2: PhantomData,
+                poseidon_config: self.poseidon_config.clone(),
+                ccs: self.ccs.clone(),
+                i: Some(self.i),
+                i_usize: Some(i_usize),
+                z_0: Some(self.z_0.clone()),
+                z_i: Some(self.z_i.clone()),
+                external_inputs: Some(external_inputs),
+                u_i_C: Some(self.u_i.C),
+                U_i: Some(self.U_i.clone()),
+                U_i1_C: Some(U_i1.C),
+                F: self.F.clone(),
+                x: Some(u_i1_x),
+                nimfs_proof: Some(nimfs_proof),
+
+                // cyclefold values
+                cf_u_i_cmW: Some(cf_u_i.cmW),
+                cf_U_i: Some(self.cf_U_i.clone()),
+                cf_x: Some(cf_u_i1_x),
+                cf_cmT: Some(cf_cmT),
+            };
+
+            // assign the next round instances
+            self.cf_W_i = cf_W_i1;
+            self.cf_U_i = cf_U_i1;
+        }
+
+        let (cs, _) = augmented_f_circuit.compute_cs_ccs()?;
+
+        #[cfg(test)]
+        assert!(cs.is_satisfied()?);
+
+        let (r1cs_w_i1, r1cs_x_i1) = extract_w_x::<C1::ScalarField>(&cs); // includes 1 and public inputs
+
+        let r1cs_z = [
+            vec![C1::ScalarField::one()],
+            r1cs_x_i1.clone(),
+            r1cs_w_i1.clone(),
+        ]
+        .concat();
+        // compute committed instances, w_{i+1}, u_{i+1}, which will be used as w_i, u_i, so we
+        // assign them directly to w_i, u_i.
+        let (u_i, w_i) = self
+            .ccs
+            .to_cccs::<_, C1, CS1>(&mut rng, &self.cs_params, &r1cs_z)?;
+        self.u_i = u_i.clone();
+        self.w_i = w_i.clone();
+
+        // set values for next iteration
+        self.i += C1::ScalarField::one();
```

Comment:
Don't we bump `i_usize` also?

Reply:
But the `AugmentedFCircuit` had it inside. And that confused me. Now I see that in any case, doesn't need to be updated for the circuit struct as per your comment above. So this is resolved.